docs/tinymist/frontend/neovim.typ

```diff
@@ -22,121 +26,81 @@ See #link("https://github.com/Myriad-Dreamin/tinymist#features")[Tinymist Featur
-
-=== Setup for #link("https://www.lazyvim.org/")[LazyVim]
-
-Copy or merge the two files to corresponding paths into `~/.config/nvim/`.
-
-- #link("https://github.com/Myriad-Dreamin/tinymist/tree/main/editors/neovim/config/autocmds.lua")[Autocmds file] will help associate the `.typ` file extension with the `typst` filetype.
-- #link("https://github.com/Myriad-Dreamin/tinymist/tree/main/editors/neovim/plugins/tinymist.lua")[Plugin file] will help start tinymist for buffers with the `typst` filetype.
-
-Then, check and restart Neovim.
-
-=== Setup for #link("https://github.com/AstroNvim")[AstroVim]
-
-AstroNvim now uses tinymist by default. Please check the #link("https://github.com/AstroNvim/astrocommunity/tree/main/lua/astrocommunity/pack/typst")[setup script].
-
-=== Setup for #link("https://github.com/neoclide/coc.nvim")[coc.nvim]
-
-You can edit the `coc-settings.json` by executing `:CocConfig`:
-
-```json
-{
-  "languageserver": {
-    "tinymist": {
-      "command": "tinymist",
-      "filetypes": ["typst"],
-      "settings": { ... }
+- Or manually:
+
+ #include "common-finding-executable.typ"
+
+== Configuration
+- With `lspconfig` 
+    ```lua
+    require("lspconfig")["tinymist"].setup {
+        settings = {
+            tinymist = {
+                settings = {
+                    formatterMode = "typstyle",
+                    exportPdf = "onType",
+                    semanticTokens = "disable"
+                    -- ...
+                },
+            },
+        },
     }
-  }
-}
-```
-
-== Tips
-<tips>
-=== Working with Multiple-File Projects
-<working-with-multiple-file-projects>
-The solution is a bit internal, which should get further improvement, but you can pin a main file by command.
-
-```lua
--- pin the main file
-vim.lsp.buf.execute_command({ command = 'tinymist.pinMain', arguments = { vim.api.nvim_buf_get_name(0) } })
--- unpin the main file
-vim.lsp.buf.execute_command({ command = 'tinymist.pinMain', arguments = { nil } })
-```
+    ```
+
+- Or with `Coc.nvim`
+
+    ```json
+    {
+      "languageserver": {
+        "tinymist": {
+          "command": "tinymist",
+          "filetypes": ["typst"],
+          "settings": { ... }
+        }
+      }
+    }
+    ```
+- Or finally with the builtin lsp protocol
+    
+    ```lua
+    vim.lsp.config['tinymist'] = {
+        cmd = {'tinymist'}, 
+        filetypes = {'typst'}
+        settings = {
+            -- ...
+        }
+    }
+    ```
+    For a full list of availible settings see #link("https://github.com/Myriad-Dreamin/tinymist/tree/main/editors/neovim/Configuration.md")[Tinymist Server Configuration]. 
 
-There is also a plan to support multiple-file projects with a workspace configuration, but I don’t know whether it is Neovim’s way, so it needs further discussion.
+== Formatting
+    Install either `typststyle` or `typstfmt` either via source or via `Mason`. Then specify the formatter in your config
```

Comment:
we don't have to install binary to get these formatter work, because they as lightweight libraries have been integrated into tinymist's language server.